src/txHashBuilder.c

```diff
@@ -257,19 +258,21 @@ void txHashBuilder_addOutput_tokenGroup(
 		{
 			BUILDER_APPEND_CBOR(CBOR_TYPE_MAP, numTokens);
 		}
-		builder->state = TX_HASH_BUILDER_IN_OUTPUTS_TOKEN;
+		builder->state = nextState;
 	}
 }
 
-void txHashBuilder_addOutput_token(
-        tx_hash_builder_t* builder,
-        const uint8_t* assetNameBuffer, size_t assetNameSize,
-        uint64_t amount
-)
+__noinline_due_to_stack__
+static void addToken(tx_hash_builder_t* builder,
+                     const uint8_t* assetNameBuffer, size_t assetNameSize,
+                     uint64_t amount,
+                     tx_hash_builder_state_t mandatoryState,
+                     tx_hash_builder_state_t nextGroupState,
+                     tx_hash_builder_state_t backoutState)
```

Comment:
Perhaps `leavingState`? I am afraid "backout" is not clear enough, or not common enough as a word.